### DISCUSSION TITLE
text normalization and preprocessing before synthesis

### DISCUSSION THREAD

**Discussion by denizsincar29** [snakers4/silero-models#232] [Q&A] [answered]
As we know, the synthesizer needs very ready and very normalized text, even with the letter yo at exact places and spelled digets. But what is the best way to automate this? say e.g. 15 см would be пятнадцать сантиметров? or 5.3 would be five point three in english? I think we need another ai for text normalizing. The synth is truely great, but, as we said, eats pre-cooked meal, doesn't cook it for itself.
Please can you advise a great ai for text normalizing and example using it with silero?
p.s. зам+ок and з+амок must also be differenciated. большие дом+а --- я д+ома.
I think it is better to make a different ai or program for preprocessor, and pass the ready intermediate code to the tts to generate speech.

**Comment by snakers4** [comment, marked as answer]
https://github.com/snakers4/silero-models/discussions/221

**Reply by denizsincar29** [reply]
but that doesn't do десятичные дроби and abbriviations. there once was a repo stt text normalization, but i couldn't run it.
